backend/src/test/java/com/engineering/thesis/backend/controllerIntegrationTests/AppointmentEPTests.java

```diff
@@ -0,0 +1,191 @@
+                ));
+        RulesMap().forEach((name, Role) -> {
+            try {
+                System.out.println("Performing select appointment with role: " + name);
+                if (name == "Doctor" || name == "Patient") {
+                    this.mockMvc
+                            .perform(MockMvcRequestBuilders.get("/api/appointment/select").with(Role))
+                            .andExpect(status().isOk())
+                            .andExpect(MockMvcResultMatchers.jsonPath("$.size()").value(2))
+                            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cost").value(2000))
+                            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cost").value(1000));
+                }
+                if (name == "Invalid") {
+                    this.mockMvc
+                            .perform(MockMvcRequestBuilders.get("/api/appointment/select").with(Role))
+                            .andExpect(status().isForbidden());
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Test
+    public void selectByIdShouldReturnAppointmentWhenProperRoleIsSelected() throws Exception {
+        final User userDoctor1 = new User(1l, "Tom", "Kowalsky", "dsadzx1xczsa@osom.com", "1I@wsdas", "ROLE_DOCTOR", true);
+        final Doctor doctor1 = new Doctor(1l, userDoctor1, "Cardiology");
+        final User userPatient1 = new User(2l, "Tom", "Kowalsky", "dsad12ssssa@osom.com", "1I@wsdas", "ROLE_PATIENT", true);
+        final Patient patient1 = new Patient(1l, userPatient1, true);
+        final Long id = 1L;
+        when(appointmentService.selectAppointmentById(id))
+                .thenReturn(java.util.Optional.of(
+                        new Appointment(1l, patient1, doctor1, 1000L, LocalDateTime.now())));
+        RulesMap().forEach((name, Role) -> {
+            try {
+                System.out.println("Performing selectById appointment with role: " + name);
+                if (name == "Doctor" || name == "Patient") {
+                    this.mockMvc
+                            .perform(MockMvcRequestBuilders.get("/api/appointment/select/1").with(Role))
+                            .andExpect(status().isOk())
+                            .andExpect(MockMvcResultMatchers.jsonPath("$.cost").value(1000))
+                            .andExpect(MockMvcResultMatchers.jsonPath("$.id").value(1L));
+                }
+                if (name == "Invalid") {
+                    this.mockMvc
+                            .perform(MockMvcRequestBuilders.get("/api/appointment/select/1").with(Role))
+                            .andExpect(status().isForbidden());
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Test
+    public void createAppointmentShouldBePossibleWhenProperRoleIsSelected() throws Exception {
```

Comment:
Names for tests are to long but explains flow. Later i'll change naming and add TC to explain each flow.

Reply:
long naming methods/tests is ok as long as it giving sense what exactly they're doing. And in my opinion you don't have to change it, because thanks this I know immediately what this test should do